crypto/asn1/x_pkey.c

```diff
@@ -89,7 +93,7 @@ void X509_PKEY_free(X509_PKEY *x)
     if (x == NULL)
         return;
 
-    i = CRYPTO_add(&x->references, -1, CRYPTO_LOCK_X509_PKEY);
+    CRYPTO_atomic_add(&x->references, -1, &i, x->lock);
```

Comment:
ret->lock could be be NULL here.
As when X509_PKEY_new failed on CRYPTO_THREAD_lock_new() call.
Please fix.